### PR TITLE
Pubsub topic infra

### PR DESCRIPTION
**Pull request by nthings**
Adding `app.pubsub_topic('topic_name')` to create a topic infrastructure. And rename `@app.topic('topic_name')` to `@app.pubsub_subscription('topic_name')` `topic` still works but it has a deprecation message now for backwards compatibility. In colaboration with @mauriciowittenberg

### CODE REVIEW

File: goblet/tests/data/http/pubsub-deploy/delete-v1-projects-test_project-topics-test_1.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+    "headers": {},
+    "body": {}
+  }
```
Comment (klahnen) [inline goblet/tests/data/http/pubsub-deploy/delete-v1-projects-test_project-topics-test_1.json:4]: chore: EOF missing

File: goblet/tests/data/http/pubsub-deploy/put-v1-projects-test_project-topics-test_1.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{   
+    "headers": {},
+    "body": {
+        "name": "projects/goblet/topics/test"
+    }
+}
```
Comment (klahnen) [inline goblet/tests/data/http/pubsub-deploy/put-v1-projects-test_project-topics-test_1.json:6]: chore: EOF missing

File: goblet/write_files.py [hunk 3]
```diff
@@ -40,7 +40,7 @@ def main(request):
 #     return jsonify("success")
 
 # pubsub topic
-# @app.topic('test_topic')
+# @app.pubsub_subscription('test_topic')
```
Comment (klahnen) [inline goblet/write_files.py:43]: question: can we remove commented code?

**Review by klahnen — CHANGES_REQUESTED** [review]

File: goblet/write_files.py (hunk 3 above)
Comment (nthings) [reply goblet/write_files.py:43]: @anovis Can this be removed?

File: goblet/decorators.py [hunk 4]
```diff
@@ -128,13 +130,29 @@ def bqremotefunction(
...
 
+    def topic(self, topic, **kwargs):
+        warn(
+            "This method is deprecated, use @app.pubsub_subscription",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+        return self.pubsub_subscription(topic, **kwargs)
+
+    def topic(self, topic, **kwargs):
```
Comment (anovis) [inline goblet/decorators.py:148]: we have duplicate code

File: goblet/infrastructures/pubsub.py [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+                self.versioned_clients.pubsub_topic.execute(
+                    "create",
+                    parent_key="name",
+                    parent_schema=resource["name"],
+                    params={"body": params},
+                )
+                log.info(f'PubSub Topic [{resource["id"]}] deployed')
+
+            except HttpError as e:
+                if e.resp.status == 409:
```
Comment (anovis) [inline goblet/infrastructures/pubsub.py:64]: we should add an update capability for the topic. 
and then add a test for the update. 

for example have a topic. and then run with a config 
```
@app.topic("pb-topic-publish-austen", config={"messageRetentionDuration":"3600s"})

```

File: goblet/tests/test_pubsub.py [hunk 6]
```diff
@@ -1,502 +1,108 @@
...
-        assert responses[0]["body"] == {}
+        assert delete_pubsub_topic["body"] == {}
+        assert pubsub_topic.resource["test"]["id"] == "test"
 
-    def test_update_pubsub_subscription(self, monkeypatch):
+    def test_update_pubsub_topic(self, monkeypatch):
         monkeypatch.setenv("GOOGLE_PROJECT", "goblet")
         monkeypatch.setenv("GOOGLE_LOCATION", "us-central1")
-        monkeypatch.setenv("G_TEST_NAME", "pubsub-update-subscription")
+        monkeypatch.setenv("G_TEST_NAME", "pubsub-deploy")
```
Comment (anovis) [inline goblet/tests/test_pubsub.py:69]: this should be a seperate folder for this test. `pubsub-update`. we can just copy the files we need over to that new folder.

Comment (nthings) [reply goblet/tests/test_pubsub.py]: done!

**Pull request merged**